src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -479,17 +465,17 @@ private void CreateFromBinaryForm(byte[] binaryForm, int offset)
                 (((long)binaryForm[offset + 6]) << 8) +
                 (((long)binaryForm[offset + 7])));
 
-            SubAuthorities = new int[binaryForm[offset + 1]];
+            int subAuthoritiesLength = binaryForm[offset + 1];
 
             //
             // Subauthorities are represented in big-endian format
             //
 
-            for (byte i = 0; i < binaryForm[offset + 1]; i++)
+            for (int i = 0; i < binaryForm[offset + 1]; i++)
             {
                 unchecked
                 {
-                    SubAuthorities[i] =
+                    subAuthorities[i] =
```

Comment:
I don't see anything that complains if `binaryForm[offset+1]` exceeds MaxSubAuthorities.  Please write a test that does so.

I think this is going to give a non-friendly out-of-bounds exception instead of an appropriate "your data is malformed"-type exception.